docs/proposals/21-ipv6-singlestack-local.md

```diff
@@ -0,0 +1,213 @@
+Corresponding instructions are made available for macOS and Linux machines.
+
+provider-local is configured to use IPv6 addresses for patching the status of LoadBalancer services.
+This is auto-detected if provider-local runs on the seed itself and configured explicitly if running on the host machine (which might have both IPv4 and IPv6 addresses).
+
+#### Scheduling
+
+gardener-apiserver only allows scheduling shoots to seeds with matching `ipFamilies` settings.
+I.e., all `Shoot.spec.networking.ipFamilies` values must be included in `Seed.spec.networks.ipFamilies`.
+This might be changed later on if "networking magic" is in place to host dual-stack shoots on IPv6 singe-stack seeds or similar.
+
+Also, gardener-scheduler chooses seeds with `ipFamilies` settings matching the shoot's configuration.
+
+#### DNS Records
+
+gardenlet creates `DNSRecord` objects for the shoot API server with the record type corresponding to the `Shoot.spec.networking.ipFamilies` setting (`A` for IPv4, `AAAA` for IPv6).
+`AAAA` records are created with a suffix to avoid name collisions in a potential dual-stack implementation.
+
+#### Network Policies
+
+Gardener manages several `NetworkPolicy` objects with static CIDRs (e.g. `allow-to-public-networks` in seeds).
+For these, IPv6-equivalents of the IPv4 CIDRs are added, e.g., `::/0` as an equivalent to `0.0.0.0/0`.
+
+#### Shoot Worker Node Kernel Configuration
+
+Kubernetes networking requires IPv6 forwarding to be enabled on the OS level.
+Hence, gardenlet explicitly enables the corresponding kernel setting for shoot worker nodes via `OperatingSystemConfigurations`, similar to IPv4 (ref [gardener/gardener#7046](https://github.com/gardener/gardener/pull/7046)).
+
+#### Docker Hub Images
+
+The `docker.io` registry doesn't support pulling images over IPv6 (see [Beta IPv6 Support on Docker Hub Registry](https://www.docker.com/blog/beta-ipv6-support-on-docker-hub-registry/)).
+
+Container images from `docker.io` used on shoots and seeds are rewritten to `registry.ipv6.docker.com` if the corresponding `ipFamilies` field specifies IPv6 singe-stack.
+
+#### E2E Tests
+
+Small changes in the OS and networking stack might lead to regressions of the IPv6 feature.
+Hence, [e2e tests](../development/testing.md#end-to-end-e2e-tests-using-provider-local) (presubmits and periodics) are added to prevent unnoticed regressions of the feature.
+
+#### Networking Extensions
+
+Networking extensions need to support the configuration of IPv6-related settings of the networking implementation in their `NetworkConfig` API.
+This is not specified any further by gardener and will be implemented differently in different extensions (as the `NetworkConfig` API design differs already today).
+In general, extensions need to respect the `Shoot.spec.networking.ipFamilies` settings in all aspects: API validation, defaulting, and corresponding handling in code (i.e., configuring IPAM, etc.).
```

Comment:
Maybe we should add a list of network extension affected here.

Reply:
As all network extensions are affected, we just would add all extensions here. Does it make more sense to list the extensions in the implementation issue #7051 ?

Reply:
you are right.